### PR TITLE
Link Hadoop YARN exploit to documentation

### PR DESCRIPTION
**Pull request by raboof**
This exploit scans for misconfigured installations, link to the documentation that describes how to properly secure it.

## Verification

List the steps needed to make sure this thing works

- [x] Start `msfconsole`
- [x] `use exploit/linux/http/hadoop_unauth_exec`
- [ ] ...
- [ ] **Verify** the thing does what it should
- [ ] **Verify** the thing does not do what it should not
- [ ] **Document** the thing and how it works ([Example](https://github.com/rapid7/metasploit-framework/blob/master/documentation/modules/post/multi/gather/aws_keys.md))

### CODE REVIEW

File: modules/exploits/linux/http/hadoop_unauth_exec.rb [hunk 1]
```diff
@@ -26,8 +26,9 @@ def initialize(info = {})
       'References'     =>
         [
           ['URL', 'http://archive.hack.lu/2016/Wavestone%20-%20Hack.lu%202016%20-%20Hadoop%20safari%20-%20Hunting%20for%20vulnerabilities%20-%20v1.0.pdf'],
-          ['URL', 'https://github.com/vulhub/vulhub/tree/master/hadoop/unauthorized-yarn']
-          # Note, there will never be a CVE for this issue, unless something radical changes in the CVE inclusion rules.
+          ['URL', 'https://github.com/vulhub/vulhub/tree/master/hadoop/unauthorized-yarn'],
+          # Note, there will never be a CVE for this issue, since this is a misconfiguration by the administrator rather than a vulnerability in the software:
+          ['URL', 'https://hadoop.apache.org/docs/current/hadoop-project-dist/hadoop-common/SecureMode.html']
```
Comment (gwillcox-r7) [inline modules/exploits/linux/http/hadoop_unauth_exec.rb:31]: This is a nice addition but it fails to explain how this ties into the rest of this exploit. All it tells a user is "use Kerberos" without explaining how the default settings are bad. It also doesn't tell me how switching to Kerberos helps things. I'd update the comment here to explain this a bit more.

File: modules/exploits/linux/http/hadoop_unauth_exec.rb [hunk 2]
```diff
@@ -14,8 +14,8 @@ def initialize(info = {})
     super(update_info(info,
       'Name'           => 'Hadoop YARN ResourceManager Unauthenticated Command Execution',
       'Description'    => %q{
-          This module uses built-in functionality to execute arbitrary commands on an unsecured Hadoop server which is not configured for strong
-          authentication, via Hadoop's standard ResourceManager REST API.
+          This module uses built-in functionality to execute arbitrary commands on Hadoop servers which are not properly configured to be exposed to
+          the network, via Hadoop's standard ResourceManager REST API.
```
Comment (gwillcox-r7) [inline modules/exploits/linux/http/hadoop_unauth_exec.rb:18]: I think the old explanation makes sense. This is about authentication mechanisms, not whether or not the server is exposed to the network or how much of its functionality is exposed. Ultimately the core issue here appears to be that you can access the ResourceManager unauthenticated. If that is exposed publicly but is protected by Kerberos authentication then your new suggested statement would be a violation of this rule, when in reality the issue described here no longer applies.

**Review by gwillcox-r7 — CHANGES_REQUESTED** [review]

**Review by gwillcox-r7 — CHANGES_REQUESTED** [review]
Looking close to done just need a bit more specifics r.e which setting is involved and the file that this is set in (core-site.xml)

File: modules/exploits/linux/http/hadoop_unauth_exec.rb [hunk 3]
```diff
@@ -14,7 +14,7 @@ def initialize(info = {})
     super(update_info(info,
       'Name'           => 'Hadoop YARN ResourceManager Unauthenticated Command Execution',
       'Description'    => %q{
-          This module uses built-in functionality to execute arbitrary commands on an unsecured Hadoop server which is not configured for strong
+          This module uses built-in functionality to execute arbitrary commands on an unsecured Hadoop server which is not configured to require
           authentication, via Hadoop's standard ResourceManager REST API.
```
[the remark this answers is not in the document]
Comment (raboof) [reply modules/exploits/linux/http/hadoop_unauth_exec.rb:18]: Makes sense to me. Will get a bit longer, perhaps something like:

> This module uses Hadoop's standard ResourceManager REST API to execute arbitrary commands on an unsecured Hadoop server. Hadoop administrators should enable Kerberos authentication for these endpoints by changing the 'hadoop.security.authentication' setting in 'core-site.xml' from 'simple' (the default) to 'kerberos' before exposing the node to the network.

WDYT?

[the remark this answers is not in the document]
Comment (gwillcox-r7) [reply modules/exploits/linux/http/hadoop_unauth_exec.rb:18]: Looks great! Happy to accept this change as it reads 👍

**Review by gwillcox-r7 — APPROVED** [review]

**Pull request merged**